src/games/supported/BattleZone.cpp

```diff
@@ -143,3 +143,36 @@ void BattleZoneSettings::loadState(Deserializer & ser) {
   m_lives = ser.getInt();
 }
 
+// returns a list of mode that the game can be played in
+ModeVect BattleZoneSettings::getAvailableModes() {
+    ModeVect modes(getNumModes());
+    for (unsigned int i = 0; i < modes.size(); i++) {
+        modes[i] = i + 1;
+    }
+    return modes;
+}
+
+// set the mode of the game
+// the given mode must be one returned by the previous function
+void BattleZoneSettings::setMode(game_mode_t m, System &system,
+    std::unique_ptr<StellaEnvironmentWrapper> environment) {
+
+    if(m == 0) {
```

Comment:
Comment on why we set this to 1 instead of throwing an exception.

Reply:
More than one game have this situation. I don't want to force people to set a mode. If they don't do it, they should get the default mode. The default mode is `0` in most games, so it was easier here to ensure this one would deal with `0` appropriately.